go/adbc/driver/flightsql/flightsql_adbc.go

```diff
@@ -324,7 +345,19 @@ func doGet(ctx context.Context, cl *flightsql.Client, endpoint *flight.FlightEnd
 }
 
 func (c *cnxn) SetOption(key, value string) error {
+	if strings.HasPrefix(key, OptionRPCCallHeaderPrefix) {
+		name := strings.TrimPrefix(key, OptionRPCCallHeaderPrefix)
+		if value == "" {
+			c.hdrs.Delete(name)
+		} else {
+			c.hdrs.Append(name, value)
+		}
+		return nil
+	}
+
 	switch key {
+	case OptionAuthorizationHeader:
```

Comment:
Do we want to accept this at the connection level?

Reply:
hmm, it's hard to say. There is the possibility that you may want to replace the authorization header? But i can't think of a good reason to do that over just making a new connection. I guess we should remove this from the connection